src/main/java/de/uni_mannheim/informatik/dws/wdi/WDI_Project_T2/util/Utils.java

```diff
@@ -63,3 +93,40 @@ public static String preprocessTransmission(String transmission) {
         return transmission;
     }
 }
+
+class WhiteListSingleton {
+    private static WhiteListSingleton single_instance = null;
+
+    public static WhiteListSingleton getInstance() {
+        if (single_instance == null)
+            single_instance = new WhiteListSingleton();
+        return single_instance;
+    }
+
+    public static Set set;
+
+    private WhiteListSingleton() {
+        ClassLoader classloader = Thread.currentThread().getContextClassLoader();
+        File file = new File(classloader.getResource("data/terms-whitelist/listfile_car_emissions.txt").getFile());
+        File file2 = new File(classloader.getResource("data/terms-whitelist/listfile_vehicles.txt").getFile());
+        try {
+            BufferedReader br = new BufferedReader(new FileReader(file));
+            BufferedReader br2 = new BufferedReader(new FileReader(file2));
+            String whiteListString = br.readLine();
+            String whiteListString2 = br2.readLine();
+            String[] whiteListArray = whiteListString.split("\\W+");
+            String[] whiteListArray2 = whiteListString2.split("\\W+");
+            this.set = new HashSet<String>();
+
+            for (String str : whiteListArray) {
+                this.set.add(str.toLowerCase());
+            }
+
+            for (String str : whiteListArray2) {
+                this.set.add(str.toLowerCase());
+            }
+        } catch (Exception e) {
+            // nothing
```

Comment:
Probably you should at least log the error and initialise the set. Otherwise it will fail with a NullPointer and no one knows why